snapshots/devmapper/pool_device.go

```diff
@@ -479,6 +481,41 @@ func (p *PoolDevice) GetUsage(deviceName string) (int64, error) {
 	return count * dmsetup.SectorSize, nil
 }
 
+func (p *PoolDevice) IsOutOfSpace(deviceName string) (bool, error) {
+	var (
+		transactionId uint64
+		dataUsed      uint64
+		dataTotal     uint64
+		metadataUsed  uint64
+		metadataTotal uint64
+		lowWaterMark  uint64
+	)
+
+	status, err := dmsetup.Status(deviceName)
+	if err != nil {
+		return false, errors.Wrapf(err, "can't get status for device %q", deviceName)
+	}
+
+	if len(status.Params) == 0 {
+		return false, errors.Errorf("failed to get the number of used blocks, unexpected output from dmsetup status")
+	}
+
+	fmt.Sscanf(strings.Join(status.Params, " "), "%d %d/%d %d/%d", &transactionId, &metadataUsed, &metadataTotal, &dataUsed, &dataTotal)
```

Comment:
How about calling strconv.Atoi() instead of fmt.Sscanf()?

Reply:
strconv.Atoi() is known to be faster than fmt.Sscanf(), but in this case, it seems to meet the need of reading 5 integers from a output of known format, with just one line of code.

Reply:
Ah, sorry. dmsetup.Status() doesn't parse the part you want to use. How about changing dmsetup.Status() instead of having a parsing logic in two parts?

Reply:
> Ah, sorry. dmsetup.Status() doesn't parse the part you want to use. How about changing dmsetup.Status() instead of having a parsing logic in two parts?

If we change dmsetup.Params() and add the parsing logic there, we will be unnecessarily parsing the Params even if caller does not use it.